### PR TITLE
Validate labels for profiles received from agent

### LINKED ISSUE #817: Validate label names
It's currently possible to write invalid label names to the database that are then impossible to query using the query language. That's how https://github.com/parca-dev/parca-agent/pull/316 could sneak in without us noticing.

### PR DESCRIPTION
**Pull request by Sylfrena**
Fixes #817

### CODE REVIEW

**Comment by Sylfrena** [comment]
@brancz, Should I have sanitized the labels instead?

File: pkg/profilestore/profilestore_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		Labels: []*profilestorepb.Label{{
+			Name:  "n1:n",
+			Value: "v1",
+		}},
+	}
+
+	samples := []*profilestorepb.RawSample{{RawProfile: buf.Bytes()}}
+
+	req := &profilestorepb.WriteRawRequest{
+		Tenant: "",
```
Comment (brancz) [inline pkg/profilestore/profilestore_test.go:67]: Slight nit: This line is not necessary as the empty string is the default value already. Also, this field is deprecated.

File: pkg/profilestore/profilestore_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		trace.NewNoopTracerProvider().Tracer(""),
+		db,
+		mStr,
+	)
+
+	buf := bytes.NewBuffer(nil)
+	ctx := context.Background()
+
+	// labelset with invalid label name
+	labelSet := profilestorepb.LabelSet{
```
Comment (brancz) [inline pkg/profilestore/profilestore_test.go:57]: A bit of a style choice but we usually inline these when we can, which is possible here.

File: pkg/profilestore/profilestore_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		Labels: []*profilestorepb.Label{{
+			Name:  "n1:n",
+			Value: "v1",
+		}},
+	}
+
+	samples := []*profilestorepb.RawSample{{RawProfile: buf.Bytes()}}
```
Comment (brancz) [inline pkg/profilestore/profilestore_test.go:64]: I think you should be able to skip this entirely, no? Validation of labels happens before samples are even looked at so we can just leave it completely empty.

**Review by brancz — COMMENTED** [review]
Nice! Yes, this is pretty much exactly what I was thinking. Just a few style choices to be consistent with the rest of the codebase.

File: pkg/profilestore/profilestore.go [hunk 4]
```diff
@@ -65,6 +66,10 @@ func (s *ProfileStore) WriteRaw(ctx context.Context, r *profilestorepb.WriteRawR
 	for _, series := range r.Series {
 		ls := make(labels.Labels, 0, len(series.Labels.Labels))
 		for _, l := range series.Labels.Labels {
+			if err := model.LabelName(l.Name).IsValid(); !err {
```
Comment (kakkoyun) [inline pkg/profilestore/profilestore.go:69]:
```suggestion
			if valid := model.LabelName(l.Name).IsValid(); !valid {
```

This is more idiomatic. `err` usually reserved for a type conforms `error` interface.

**Review by brancz — COMMENTED** [review]
Nice! lgtm 🎉

**Pull request merged**